ReactWindows/ReactNative.Shared/Bridge/IReactDelegateFactory.cs

```diff
@@ -21,8 +21,9 @@ public interface IReactDelegateFactory
         /// </summary>
         /// <param name="nativeModule">The native module instance.</param>
         /// <param name="method">The method.</param>
+        /// <param name="genericDelegate">The delegate used to avoid expensive reflection invocation.</param>
         /// <returns>The invocation delegate.</returns>
-        Action<INativeModule, IReactInstance, JArray> Create(INativeModule nativeModule, MethodInfo method);
+        Action<INativeModule, IReactInstance, JArray> Create(INativeModule nativeModule, MethodInfo method, IGenericDelegate genericDelegate);
```

Comment:
>IGenericDelegate [](start = 108, length = 16)

The `IGenericDelegate` is created using the `INativeModule` and the `MethodInfo`. Can we remove the `IGenericDelegate` from this interface? We should be able to push the factory call into the `GenericDelegateFactory` implementation.